### PR TITLE
Define the "jamf_service" configuration

### PR DESCRIPTION
**Pull request by codingllama**
Add the "jamf_service" stanza to fileconf and the corresponding entry to `servicecfg`.

Changes on `servicecfg`, `enterpriseServicesEnabled` and `ApplyFileConfig` are part of the groundwork for the Jamf MDM service, as is the configuration itself.

https://github.com/gravitational/teleport.e/issues/826

### CODE REVIEW

File: lib/config/fileconf.go [hunk 1]
```diff
@@ -2385,3 +2388,101 @@ type Okta struct {
...
+
+// JamfService is the yaml representation of jamf_service.
+// Corresponds to [types.JamfSpecV1].
+type JamfService struct {
+	Service `yaml:",inline"`
+	// Name is the name of the sync device source.
+	Name string `yaml:"name,omitempty"`
+	// SyncDelay is the initial sync delay.
+	// Zero means "server default", negative means "immediate".
+	SyncDelay *time.Duration `yaml:"sync_delay,omitempty"`
```
Comment (mdwn) [inline lib/config/fileconf.go:2400]: Do the durations need to be pointers?

File: lib/config/fileconf.go [hunk 2]
```diff
@@ -2385,3 +2388,101 @@ type Okta struct {
...
+	if pwd == "" {
+		return nil, trace.BadParameter("jamf password_file is empty")
+	}
+	// Trim trailing \n?
+	if l := len(pwd); pwd[l-1] == '\n' {
+		pwd = pwd[:l-1]
+	}
+
+	// Assemble spec.
+	fromNilDuration := func(d *time.Duration) types.Duration {
```
Comment (mdwn) [inline lib/config/fileconf.go:2457]: If the durations don't need to be pointers, I suspect you can remove this bit entirely.

**Review by mdwn — APPROVED** [review]
Overall LGTM, just one question w/r/t the time.Duration pointers.

File: lib/config/fileconf.go (hunk 1 above)
Comment (codingllama) [reply lib/config/fileconf.go:2400]: The idea was to be able to tell apart set from unset, but I suppose that doesn't make a difference anymore. Done.

File: lib/config/fileconf.go (hunk 2 above)
Comment (codingllama) [reply lib/config/fileconf.go:2457]: Done.

**Comment by codingllama** [comment]
Friendly ping @strideynet @hugoShaka @probakowski ?

**Review by hugoShaka — APPROVED** [review]
Sorry for the delay.

The RFD-7e is extremely clear and detailed. It gives all the context needed for the review. Thank you, this really helps :)

**Comment by codingllama** [comment]
Thanks for the kind words, Hugo.

**Review by strideynet — APPROVED** [review]

**Pull request merged**